File: 09. JS Basic/02. Loops and Functions/finder-simple-numbers.js
function check_simple(number) {

    if (number > 1) {
        for (let i = 2; i < number/2 + 1; i++) {
            if (number % i === 0) {
                return false
            }
        }
        return true
    } else {
        return false
    }
}

function func(n) {
    let arr = []
    let i = 2

    while (arr.length < n) {
        if (check_simple(i)) {
            arr.push(i)
        }
        i++
    }
    return arr
}

console.time()
console.log(func(process.argv[2]))
console.timeEnd()

File: 09. JS Basic/02. Loops and Functions/finder-simple-numbers.py

import time


class Solution(object):
    def timeit(wrapped_func):
        def wrap_func(*args, **kw):
            t = time.time_ns()
            result = wrapped_func(*args, **kw)
            print('Processing time of %s(): %.2f seconds.' %
                  (wrapped_func.__qualname__, time.time_ns() - t))
            return result

        return wrap_func

    @timeit
    def func(self, n):
        def check_simple(number):
            if number > 1:  
                for i in range(2, int(number/2) + 1):  
                    if number % i == 0:  
                        return False
                return True
            else:
                return False

        arr = []
        i = 2

        while len(arr) < n:
            if check_simple(i):
                arr.append(i)
            i += 1

        return arr


if __name__ == '__main__':
    obj = Solution()
    print(obj.func(5))
